packages/eslint-plugin/src/rules/no-unnecessary-coercion.ts

```diff
@@ -0,0 +1,242 @@
+        node: TSESTree.Expression,
+      ): void {
+        const memberExpr = node.parent as TSESTree.MemberExpression;
+        const type = getConstrainedTypeAtLocation(services, memberExpr.object);
+        if (doesUnderlyingTypeMatchFlag(type, ts.TypeFlags.StringLike)) {
+          context.report({
+            node,
+            messageId: 'unnecessaryCoercion',
+            fix: (fixer): RuleFix[] => [
+              fixer.removeRange([
+                memberExpr.parent.range[0],
+                memberExpr.object.range[0],
+              ]),
+              fixer.removeRange([
+                memberExpr.object.range[1],
+                memberExpr.parent.range[1],
+              ]),
+            ],
+            loc: {
+              start: memberExpr.object.loc.end,
+              end: memberExpr.parent.loc.end,
+            },
+            data: {
+              violation: 'Using .toString() on a string',
+              type: 'string',
+            },
+          });
+        }
+      },
+      'AssignmentExpression[operator = "+="], BinaryExpression[operator = "+"]'(
+        node: TSESTree.AssignmentExpression | TSESTree.BinaryExpression,
+      ): void {
+        const leftType = services.getTypeAtLocation(node.left);
+        const rightType = services.getTypeAtLocation(node.right);
+        if (
+          doesUnderlyingTypeMatchFlag(leftType, ts.TypeFlags.StringLike) &&
+          node.right.type === AST_NODE_TYPES.Literal &&
+          node.right.value === ''
+        ) {
+          context.report({
+            node,
+            messageId: 'unnecessaryCoercion',
+            fix: (fixer): RuleFix[] => [
+              fixer.removeRange([node.range[0], node.left.range[0]]),
+              fixer.removeRange([node.left.range[1], node.range[1]]),
+            ],
+            loc: {
+              start: node.left.loc.end,
+              end: node.loc.end,
+            },
+            data: {
+              violation: "Concatenating a string with ''",
+              type: 'string',
+            },
+          });
+        }
+        if (
+          node.left.type === AST_NODE_TYPES.Literal &&
+          node.left.value === '' &&
+          doesUnderlyingTypeMatchFlag(rightType, ts.TypeFlags.StringLike)
+        ) {
+          context.report({
+            node,
+            messageId: 'unnecessaryCoercion',
+            fix: (fixer): RuleFix[] => [
+              fixer.removeRange([node.range[0], node.right.range[0]]),
+              fixer.removeRange([node.right.range[1], node.range[1]]),
+            ],
+            loc: {
+              start: node.loc.start,
+              end: node.right.loc.start,
+            },
+            data: {
+              violation: "Concatenating '' with a string",
+              type: 'string',
+            },
+          });
+        }
+      },
+      'UnaryExpression[operator = "+"]'(node: TSESTree.UnaryExpression): void {
```

Comment:
I see what you mean. Not really sure. The error message I landed on makes me feel like I'm ok with still flagging these:

```
Using the unary + operator on a number does not change the type or value of the number
```

I'd assume the reason people would want to verbosely declare positive numbers would commonly have to do with having positive and negative counterparts for a number, eg `+100` and `-100`. Imagine then:

```
const PI = 3.14159
const positive_PI = +PI
const negative_PI = -PI
```

Now they're not using a number literal, but your reasoning still applies, they're just being verbose. Feels like a bit of a slippery slope? I would personally be using my rule (in line with what I generally expect from a linter) to tell me if I'm writing symbols that don't do anything. But it could be hard for a user in that context to see what that has to do with type conversion.